src/Routers/UsersRouter.js

```diff
@@ -418,6 +458,9 @@ export class UsersRouter extends ClassesRouter {
     this.route('POST', '/login', req => {
       return this.handleLogIn(req);
     });
+    this.route('POST', '/createSessionWithoutAuthCheck', req => {
```

Comment:
I would not call that `withoutAuthCheck` but a more common name for this process which is `impersonate`

So maybe simply `impersonateUser`

Reply:
Sure, I'd be happy to change it. 

`impersonateUser` would give me the impression that I wouldn't need to feed the response value into a separate `Parse.User.become(...)`. Do you think that interpretation could be common?

Reply:
I think the commonly understood term of "user impersonation" goes beyond the scope of this endpoint. Maybe we can find a more specfic term.

Reply:
> There is no REST API endpoint /createSession, instead the more descriptive /login term is used. So whatever the endpoint for this is named, it makes sense to begin with /login..., to indicate its functional relation.

Should this feature be semantically comparable to a normal login? Specifically, the login handler has before/after hooks, which may not be appropriate here. Imagine an API endpoint performing a "login" every time an API call is made, where the system is configured with an afterLogin hook to email the user "we saw a login from a new location". Is that appropriate system behavior? If not, is the answer to omit the hooks from this new form of login and somehow signal to developers that two things named "login" aren't really the same, or to add a parameter to skip over the hooks?

Reply:
That's a good point. How about using the `/loginAs` endpoint to give some differentiation and still indicate its relation to login? Then in all the `login` triggers we could add a parameter that indicates how the login happened, if that differentiation is not already possible.

Reply:
Alright. I will:

- Rename the new REST endpoint to /loginAs
- Update the endpoint to return a `User`, same as /login
- Otherwise retain the existing behavior of the handler in the PR
- Update the tests accordingly

Reply:
Sounds good

Reply:
Awesome !